pgrx/src/iter.rs

```diff
@@ -155,6 +158,56 @@ impl<'a, T> Iterator for TableIterator<'a, T> {
     }
 }
 
+unsafe impl<'iter, C> SqlTranslatable for TableIterator<'iter, (C,)>
+where
+    C: SqlTranslatable + 'iter,
+{
+    fn argument_sql() -> Result<SqlMapping, ArgumentError> {
+        Err(ArgumentError::Table)
+    }
+    fn return_sql() -> Result<Returns, ReturnsError> {
+        let vec = vec![match C::return_sql() {
+            Ok(Returns::One(sql)) => sql,
+            Ok(Returns::SetOf(_)) => return Err(ReturnsError::TableContainingSetOf),
+            Ok(Returns::Table(_)) => return Err(ReturnsError::NestedTable),
+            err => return err,
+        }];
+        Ok(Returns::Table(vec))
+    }
+}
+
+impl<C: IntoDatum> IntoHeapTuple for (C,) {
+    unsafe fn into_heap_tuple(self, _tupdesc: pg_sys::TupleDesc) -> *mut pg_sys::HeapTupleData {
+        unimplemented!()
+    }
+}
+
+unsafe impl<C> ReturnShipping for (C,)
+where
+    C: ReturnShipping, // so we support TableIterator<'a, (Option<T>,)> as well
```

Comment:
This is... not ideal, as it traps us in a position where we might accidentally try to return something like `TableIterator<'a, (TableIterator<'b, TupleA>, TableIterator<'c, TupleB>)>`.

Reply:
This would be preferable, as it closes off that route:
```suggestion
    C: RetPackage,
```

Reply:
I should note that I'm pretty sure that this is currently impossible anyways even with this PR because the schema generator chokes on it, I just would rather it didn't require "the schema generator chokes on it" in order to stop us from doing something silly.